torchtune/modules/kv_cache.py

```diff
@@ -91,18 +97,14 @@ def update(
                 f", but found new key tensors with batch size {k_val.shape[0]}!"
             )
 
-        if (self.size + seq_len) > self.k_cache.shape[2]:
-            raise ValueError(
-                f"The current cache has been setup with a sequence length of {self.k_cache.shape[2]}"
-                f", but the cache has reached a sequence length of {(self.size + seq_len)}!"
-            )
-        cache_pos = torch.arange(self.size, self.size + seq_len, device=k_val.device)
-        self.size += seq_len
-
+        assert (self.cache_pos[0] + seq_len) <= self.k_cache.shape[2]
```

Comment:
i dont 100% understand why, but in previous PRs i was instructed to use ValueError instead of assert. I dont know if this is worth an exception. Probably because assertion msg is not a good as value error

Reply:
The error message without the assertion is even worse (raising like we did before was ideal)

Reply:
The message was worse or ValueError is worse? If its ok to use ValueError, can we keep it and use a nicer msg instead?

Reply:
Sorry, there can't be any branching logic here because dynamo will complain about control flow (this includes checking the value of a tensor) so the plain assertion error is the best we can do.